src/stream.js

```diff
@@ -29,7 +29,8 @@ async function streamPayment ({
   if (stream.isOpen()) {
     stream.end()
   }
-
+  // Wait for the stream 'end' event to be emitted so the stream can finish sending funds
+  await new Promise(resolve => stream.once('end', resolve))
```

Comment:
can we move this into the `stream.isOpen()` block? Otherwise the stream may already be closed and this will hang